### DISCUSSION TITLE
Are Material Design icons broken?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#2348] [Q&A] [answered]
I'm in the process of rewriting my docs and as a part of this did I like to use some icons from Material Design.

But weirdly enough do some specific icons not work, while others do.

As an example does `:material-sync:` display the sync icon as displayed on the Material Design page, but `:material-warning:` is not rendering.

I use Material for MkDocs v7.0.1 with PyMdownx and the MkDocs-Material Extension to allow usage of SVG icons. As mentioned does one of the above work, so the Material icons are recognized although only partially it seems.

**Comment by facelessuser** [comment]
Have you tried searching to see if the icon name exists here? https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/

The Material icon package may have changed some names in a recent version.

**Reply by Andre601** [reply]
I did and it doesn't appear on it.
But weirdly enough does the linked Material Icons page list the icon when searching for it.

**Comment by squidfunk** [comment, marked as answer]
It seems `:material-warning:` was renamed to `:material-alert:` in the upstream package:

<img width="711" alt="Bildschirmfoto 2021-02-25 um 18 21 12" src="https://user-images.githubusercontent.com/932156/109191287-4caae500-7796-11eb-9545-785e311f3978.png">

**Reply by Andre601** [reply]
Okay. That makes sense. Was just confused as the search for `warning` on the upstream page showed the icon. Probs an alias or tag for it.

**Reply by squidfunk** [reply]
They have [aliases](https://raw.githubusercontent.com/Templarian/MaterialDesign/master/meta.json) defined:

``` json
{
        "id": "9FF7DF3F-D575-433B-9C9A-69C58363C186",
        "name": "alert",
        "codepoint": "F0026",
        "aliases": [
            "warning",
            "report-problem"
        ],
        "tags": [
            "Alert \/ Error"
        ],
        "author": "Google",
        "version": "1.5.54"
    },
```

If we can find a way to make this work with the theme, I’m happy to integrate it. @facelessuser any thoughts?

**Reply by facelessuser** [reply]
Currently, we just scan the Material icon names. Doing this we build a table of names. Keep in mind, this is a generic way that works with Material, Octicons, etc. If we parsed whatever metadata you are showing here, we could certainly be aware of aliases, but then I would have to be in sync with whatever Material is currently using...unless Material bundled the meta data in itself that I could parse at runtime. Basically, I need to be in sync with whatever Material has, and then I need to add special logic specifically for Material icons.

**Reply by squidfunk** [reply]
Thanks for sharing your thoughts! I agree that it would solely cater to the Material icons and would probably complicate the custom icon integration process. Maybe it would make sense to reintegrate the extensions with Material for MkDocs at some point again if it would be possible to reach a deeper integration, but I wouldn't consider this a high-priority task.

**Reply by facelessuser** [reply]
Yup, the alternative is that I just use the latest meta data file and pre generate a list of aliases and such and just say the plugin supports icons version X. And everytime material bumps the icons, I would need to refresh the list. This is less ideal though.

Currently, I would make a note that the extension uses the physical icon name, not their generated list of friendly names.
